### DISCUSSION TITLE
Contributing to Outreachy  - summer 2022

### DISCUSSION THREAD

**Discussion by pgleeson** [ModECI/MDF#213] [General]
Thanks to all for your interest in the Outreachy proposal for this summer.

For some suggested tasks related to making contributions please see the information at https://www.outreachy.org/outreachy-may-2022-internship-round/communities/modeci/#create-jupyter-notebooks-demonstrating-machine-lea. In short, try installing the MDF package locally and running the notebook example. 

I'll be adding more information here over the next few days on issues which could be good to start with.

**Comment by victorabba** [comment]
Hi everyone, my name is Victor. I'm a python developer with a specialty in data science and machine learning. I look forward to working with you all whilst making meaningful contributions to the project.

**Reply by Monsurat-Onabajo** [reply]
Hello Everyone, my name is Onabajo Monsurat, I will like to contribute to this open source project  as  it is the prerequisite for getting the summer outreachy internship 2022.

**Comment by 29riyasaxena** [comment]
Hello! I'm Riya Saxena and I'm a second year undergraduate at the Indian Institute of Technology, Roorkee, Uttarakhand, India. I'm currently learning machine learning. Looking forward to learn with you all!

**Comment by shivani-sangwan** [comment]
Hi everyone, this side Shivani Sangwan. I have good experience with Python and ML and believe would be able to give my best to the project.

**Comment by AfiDede** [comment]
Hello everyone, I am Afi Blebo, I am an outreachy applicant and interested in the "Create Jupyter notebooks demonstrating machine learning workflows using MDF" project. 
I look forward to working with our mentors @pgleeson and Ankur Sinha and learning from you all.

**Comment by amitha6** [comment]
Hi, Iam Amitha a third year undegraduate student interested in Machine Learning. Looking forward to learn and contribute to the project with you all.

**Comment by EjuluWilson** [comment]
Hello!, I am Wilson, a computer engineer with ML and python programming experience. It will be such a lovely opportunity to work with all of you in contributing to this project. cheers!

**Comment by shivani6320** [comment]
Hello Everyone!! I'm Shivani - an outreachy applicant🤓. I am super excited to be part of this project and learn from the mentors and fellow community members. Looking forward to making meaningful contributions.✨

**Comment by sakshikaushik717** [comment]
Hi, I am Sakshi Kaushik, I have completed the above steps. I have installed the package and also ran the notebook example. I am currently working on a few issues and have one pull request open to add installation documentation (#216). Are the any other issues that I can work on?

**Comment by 29riyasaxena** [comment]
Hi! I am a beginner at open-source, and I'm facing some issues in installing the package. I have cloned the package, and I don't know what to do after it...like how to setup the setup.py file. Need urgent help regarding it. @pgleeson @sanjayankur31

**Reply by sanjayankur31** [reply]
Hi @29riyasaxena , please take a look at #212 . Steps and discussion around installing are given there.

**Reply by 29riyasaxena** [reply]
Thanks alot! It worked for me.

**Comment by FatimaArshad-DS** [comment]
Has anyone started contributing? I'm confused about how to commit to an issue. Can you point me to a useful tutorial? I have generated one PR but it seems to have created a new issue rather than pointing to an existing issue.
Thanks in advance :)

**Reply by pgleeson** [reply]
Thanks for your pull request @FatimaArshad-DS. I'll review that in the next day or two. 

The simplest way to link this to the issue is put the hashtag #208 in a commit message for the pull request, and it will show up in the issue itself. 

Another useful habit is to create a branch in your repository specifically for the issue, e.g. either feature/208 or bugfix/208 (see https://github.com/OpenSourceBrain/OSBv2/branches), and deal with fixing one issue in each branch. When the PR is merged, then you can delete the branch and keep everything tidy.

**Reply by shivani6320** [reply]
Hello @pgleeson @sanjayankur31 , I have a suggestion  you'd like to consider -- I believe we should have a **Contributing Guide** to help all the contributors which also makes it easier for mentors to keep track of all contributions by applicants.

I've created this [issue](https://github.com/ModECI/MDF/issues/221) and created a [PR](https://github.com/ModECI/MDF/pull/223) to review. Kindly have a look at it.I would be happy to consider your views on it. Thank you.

**Comment by 29riyasaxena** [comment]
Hi! I am having trouble understanding things. I have already run the examples. I want to proceed with the conversion thing but not getting a start on it. Can you please guide me on how to study and proceed with it? @pgleeson @sanjayankur31

**Reply by pgleeson** [reply]
@29riyasaxena Hopefully the comment below gives some useful next steps.

**Reply by 29riyasaxena** [reply]
Yes! Thanks Alot! :)

**Comment by pgleeson** [comment]
Hi,
I’ve added 2 suggested tasks for people interested in the project. Have a look there for possible contributions you could make to mention in your application. 
https://github.com/ModECI/MDF/discussions/224
https://github.com/ModECI/MDF/discussions/225

Please note too as general points:

- We are a relatively new software package, so there are not too many small open issues to work on (as of yet). Many of the [open issues that are currently there](https://github.com/ModECI/MDF/issues) will require an in depth knowledge of the MDF language and reference implementation, and are probably not suitable for new contributors as starter issues. We will endeavour to add more beginner level issues though.
- If any of the READMEs or documentation are unclear and you figure out a better way to say it, please update the appropriate source with a pull request.
- The accepted candidate for the eventual project will require good prior experience in Python programming and Jupyter technologies as well as hands on experience with machine learning frameworks. So far we have ~30 interested contributors for one intern post. We hope to have more posts in future rounds which will require different skill sets.

**Reply by shivani6320** [reply]
Thank You for information. That's helpful.

**Reply by Fifi-Banjoko** [reply]
Okay, noted. Thank you.

**Reply by 29riyasaxena** [reply]
Thanks alot!

**Reply by AfiDede** [reply]
> Hi, I’ve added 2 suggested tasks for people interested in the project. Have a look there for possible contributions you could make to mention in your application. #224 #225
> 
> Please note too as general points:
> 
> * We are a relatively new software package, so there are not too many small open issues to work on (as of yet). Many of the [open issues that are currently there](https://github.com/ModECI/MDF/issues) will require an in depth knowledge of the MDF language and reference implementation, and are probably not suitable for new contributors as starter issues. We will endeavour to add more beginner level issues though.
> * If any of the READMEs or documentation are unclear and you figure out a better way to say it, please update the appropriate source with a pull request.
> * The accepted candidate for the eventual project will require good prior experience in Python programming and Jupyter technologies as well as hands on experience with machine learning frameworks. So far we have ~30 interested contributors for one intern post. We hope to have more posts in future rounds which will require different skill sets.

Thank you, that is helpful!

**Comment by tejaswari7** [comment]
Hello everyone I'm Tejaswari Bhamare, I'm outreachy applicant. I have experience with python, jupyter notebook and little bit with flask. Looking forward to contribute!!

is it okay to be on Windows OS for this project?

**Reply by shivani6320** [reply]
yes @tejaswari7 , I'm also using Windows OS & its completely fine.

**Reply by tejaswari7** [reply]
Thank you for clarification!!

**Comment by DeejaKamran** [comment]
Heyy everyone! Deeja here. I am so excited for this project and for the learning experience while interacting with this community. I have experienced with some machine learning projects. I will start working on some contributions. Looking forward to your guidance 😀😀

**Comment by amina-786** [comment]
Hi, Every One. Best of luck everyone. I have done different Ml projects and having good knowledge of python. I am applicant of outreachy.

**Comment by juliaochoms42** [comment]
Hi everyone, I am called Tayo Julia. I have good experience with Python and ML and also
I have been working with Jupyter Notebook for some time now believe would be able to give my best to the project.

**Reply by Fifi-Banjoko** [reply]
Welcome @juliaochoms42!

**Reply by juliaochoms42** [reply]
> Welcome @juliaochoms42!

Thank you

**Reply by juliaochoms42** [reply]
> Welcome @juliaochoms42!

Please I wish to ask if the mentor has given out specific task to be carried out already

**Reply by sanjayankur31** [reply]
Yes, please see #224 and #225

**Comment by anamika-yadav99** [comment]
Hi everyone! I'm Anamika. I'm 3rd year undergraduate student from India.  I'm a Python Developer with open source experience working in ML, bioinformatics, ML toolboxes/packages . I'm a student researcher working in application of ML algorithms to solve problems in biotechnology.  Just out of curiosity, I have taken a few basic courses in neuroscience and quantum computing as well. This project is very exciting for me as I was looking for opportunities to explore neuroscience and AI further.

**Reply by sanjayankur31** [reply]
Welcome @anamika-yadav99 --- please take a look at #224  and #225 to get started.

**Comment by glencode** [comment]
Hello, I'm  Glen, i'm starting out my ML/datascience journey and i'm excited to be able to contribute to MDF

**Reply by pgleeson** [reply]
Welcome @glencode! See above for suggested tasks.

**Comment by parikshit14** [comment]
Hi Team,

I am Parikshit Singh Rathore, I am a senior year undergraduate studying computer science and engineering. I have been contributing to open-source for a year and a half and would love to contribute & be part of this ModECI team through outreachy.

Thanks

**Reply by parikshit14** [reply]
Almost forgot to introduce myself 😅

**Reply by Fifi-Banjoko** [reply]
Welcome Parikshit! See above for suggested tasks.

**Reply by parikshit14** [reply]
thanks Fifi, I have completed the tasks already though some changes are required, just forgot to introduce myself in the discussions section before!!

**Reply by Fifi-Banjoko** [reply]
Haha okay.

**Comment by pgleeson** [comment]
A big thank you to all of the contributors here over the past number of weeks! This has been our first Outreachy experience as a project and we've been delighted with the levels of interaction and enthusiasm.

We've had a high number of applicants, and unfortunately we will probably just have one internship place. We will certainly try to continue this in subsequent rounds and also extend Outreachy to other projects we're working on in neuroscience and ML.

Thank you all again!

**Reply by Fifi-Banjoko** [reply]
Thank you @pgleeson and @sanjayankur31 for the opportunity to contribute to the community.

**Reply by 29riyasaxena** [reply]
Thanks @pgleeson @sanjayankur31 for being so helpful in this contribution phase and correcting things again and again, I am very grateful for this.  Thanks for providing this opportunity.

**Reply by parikshit14** [reply]
The timely feed-backs and the wholesome experience were great, special thanks to @davidt0x @pgleeson @sanjayankur31 for finding out improvements when and where possible, and for the support when we got stuck.

**Reply by somyagr** [reply]
@pgleeson @sanjayankur31 Thanks for the amazing guidance throughout the contribution period, learnt a lot.

**Comment by somyagr** [comment]
Hi @pgleeson @sanjayankur31 , is there any task that I can do?

**Reply by sanjayankur31** [reply]
Hi @somyagr : the initial contribution period for Outreachy has now ended and you've all submitted your applications. So, now we wait until the candidates are announced to proceed with more tasks.

**Comment by Yusrah23** [comment]
Hi, everyone. My name is Yusrah, I'm currently learning ML and I am really excited to be part of this project. I hope everyone is looking forward to my contributions as I am to everyone's own.

**Reply by kusanele** [reply]
Hey @Yusrah23 , Looking forward to leaning with you.

**Comment by risper25** [comment]
Hello everyone, Am Risper, I have some knowledge of AI(PyTorch) and would like to contribute to the MDF project.Thanks.

**Comment by kusanele** [comment]
Good day everyone , My name is Kusanele . I am really looking forward to contributing to this project for summer internship 2023. @pgleeson @sanjayankur31 i cant wait to embark on this journey with you and everyone else.

**Comment by sanjayankur31** [comment]
@kusanele @risper25 @Yusrah23  (and all other Outreachy 2022 Winter candiates): welcome! please take a look at https://github.com/ModECI/MDF/discussions/346 for the initial tasks and so on.

**Reply by kusanele** [reply]
Will Check it out. Thank you

**Reply by risper25** [reply]
thanks @sanjayankur31
